packages/ckeditor5-engine/src/controller/datacontroller.js

```diff
@@ -541,10 +542,16 @@ function _getMarkersRelativeToElement( element ) {
 	const elementRange = ModelRange._createIn( element );
 
 	for ( const marker of doc.model.markers ) {
-		const intersection = elementRange.getIntersection( marker.getRange() );
+		const markerRange = marker.getRange();
 
-		if ( intersection ) {
-			result.push( [ marker.name, intersection ] );
+		const isMarkerCollapsedAtElementBoundary = markerRange.isCollapsed && ( markerRange.start.isAtStart || markerRange.start.isAtEnd );
```

Comment:
```suggestion
		const isMarkerCollapsedAtElementBoundary = markerRange.isCollapsed && ( markerRange.start.isEqual( elementRange.start) || markerRange.start.isEqual( elemenetRange.end ) );
```

Reply:
It's worth adding tests for the previous implementation - with a collapsed marker that is at the start or end of some other element (not included in the checked element) and for a collapsed marker at the start or end inside some deeper nested element.

Reply:
Added.